### PR TITLE
Naming, viewMananger delegation, and guide tool paperLayer handling fixes

### PR DESCRIPTION
**Pull request by MurrayLF**
- Naming so that all layers, renderLayers, components, and connections receive unique names.
- Replacement of Registry.viewManager with viewManagerDelegates.
- Creation of a paperFeatureRegistry within paperView, enabling paperFeatures to be returned to their correct layer after being removed from them for display purposes in Ron's guide tool. Modification's to the necessary functions to implement this.

### CODE REVIEW

File: src/app/core/device.ts [hunk 1]
```diff
@@ -82,6 +82,15 @@ export default class Device {
...
+    /**
+     * Sets the map used for naming objects
+     * @param {Map<string, number>}
+     * @memberof Device
+     */
+    set nameMap(nameMap: Map<string, number>) {
+        this.__nameMap = nameMap;
+    }
+
     /**
```
Comment (rkrishnasanka) [inline src/app/core/device.ts:94]: What objects are we naming here ?

File: src/app/view/viewManager.js [hunk 2]
```diff
@@ -172,6 +172,49 @@ export default class ViewManager {
...
     }
 
+    /**
+     * Sets the initial state of the name map
+     * @memberof ViewManager
+     * @returns {void}
+     */
+    setNameMap() {
+        const newMap = new Map();
+        for (let i = 0; i < this.currentDevice.layers.length; i++) {
```
Comment (rkrishnasanka) [inline src/app/view/viewManager.js:182]: I guess I just don't know what the nameMap is and how we can automatically set in the device object's lifecycle

**Review by rkrishnasanka — APPROVED** [review]
We need to revisit the `nameMap` but otherwise, everything looks good

File: src/app/view/viewManager.js (hunk 2 above)
Comment (MurrayLF) [reply src/app/view/viewManager.js:182]: nameMap simply records what the highest number applied to a given name is (e.g.; "Port": 6, "Channel": 15). nameMap is used by Device.generateNewName(), which sees if a string is a key in nameMap to get a number which enables the next name. setNameMap() is used to set the initial state of the nameMap when a JSON file is loaded in.

**Pull request merged**